src/model/events-model.js

```diff
@@ -1,17 +1,50 @@
+import Observable from '../framework/observable.js';
 import {generateEvent} from '../mock/event.js';
 
-export default class EventsModel {
-  #destinations = null;
-  #offers = null;
+export default class EventsModel extends Observable{
+  #events = Array.from({length: 20}, generateEvent);
 
-  constructor(destinationsСatalog, offersСatalog) {
-    this.#offers = offersСatalog;
-    this.#destinations = destinationsСatalog;
+  get events () {
+    return this.#events;
   }
 
-  #events = () => Array.from({length: 20}, () => generateEvent(this.#destinations, this.#offers));
+  updateEvent = (updateType, updatedEvent) => {
+    const updatedEventIndex = this.#events.findIndex((event) => event.id === updatedEvent.id);
 
-  get events () {
-    return this.#events();
-  }
+    if (updatedEventIndex === -1) {
+      throw new Error('Can\'t update unexisting event');
```

Comment:
Лучше сразу в исключение добавлять данные которые привели к его появлению. Так можно будет быстрее отладить
```suggestion
      throw new Error(`Can't update unexisting event, updatedEvent: ${updatedEvent}`);
```